Automattic/wp-calypso
# Pages: Add "Set as Homepage" action to the pages list

#### Changes proposed in this Pull Request

Complete the work started in #24087 that allows setting a page as homepage directly from the pages list.

![Screenshot 2019-08-05 at 18 52 52](https://user-images.githubusercontent.com/2070010/62484827-ef65d200-b7b2-11e9-9259-f8f3d44fba51.png)

Previously, this PR took advantage of the site settings infrastructure by adding the front page settings to the settings endpoint.
Though, this would have meant waiting for the next JP release, and adding version checks.

Now, instead, this PR uses the `sites/:site/homepage` endpoint, which was already available.
The request is sent through the data layer as usual, and the response is then used to update the `siteSettings` store.

For this reason, I've updated the front page selectors to take advantage of the`site-settings` store, instead of relying to the site options which aren't kept updated.

#### Follow Ups

- Since this changes the selectors to look into `siteSettings` instead of `sites`/`pages`.
This means that we should move them away from there and, possibly, into the generic `selectors` folder.
- Once #35106 lands, we should use the new `isSiteUsingFullSiteEditing` selector to hide the unnecessary `BlogPostsPage` component.
- Last but not least, we should also add a "Set as Blog Page" action to the page list (not available for FSE sites), which would take advantage of this PR.

#### Testing instructions



* Navigate to `/pages/:site`.
* In the ellipsis menu of a page (that is neither the homepage nor the blog page), locate the "Set as Homepage" action, and click it.
* Make sure that page immediately becomes the site's homepage (notice the label in the page card, and check on the frontend of the site).
* Make sure the "Set as Homepage" action is only available for published pages, and unavailable for the current homepage and blog page.

Fixes #35043
